hub/models.py

```diff
@@ -199,53 +199,120 @@ class Meta:
             ("order_and_feature", "Can change sort order and mark as featured")
         ]
 
+    @property
+    def value_col(self):
+        if self.is_date:
+            return "date"
+        elif self.is_float:
+            return "float"
+        elif self.is_number:
+            return "int"
+        elif self.is_json:
+            return "json"
+        else:
+            return "data"
+
     def filter(self, query, **kwargs):
         return Filter(self, query).run(**kwargs)
 
+    shades = [
+        "#ffffd9",
+        "#edf8b1",
+        "#c7e9b4",
+        "#7fcdbb",
+        "#41b6c4",
+        "#1d91c0",
+        "#225ea8",
+        "#253494",
+        "#081d58",
+    ]
+
+    def shade(self, val, cmin, cmax):
+        try:
+            x = float(val - cmin) / (cmax - cmin)
+        except ZeroDivisionError:
+            x = 0.5  # cmax == cmin
+
+        shade = int(x * 9) - 1
+        return self.shades[shade]
+
     def colours_for_areas(self, areas):
-        values = self.shader_value(areas)
+        values, mininimum, maximum = self.shader_value(areas)
+        legend = {}
+        for option in self.options:
+            if option.get("shader", None) is not None:
+                legend[option["title"]] = option["shader"]
+
+        if len(legend) > 0:
+            props = {"properties": {"legend": legend}}
+        else:
+            props = {
+                "properties": {
+                    "maximum": maximum,
```

Comment:
I think these values are what gets passed through to `explore.html` in `${ key.minimum }` and `${ key.maximum }` … if so, is there any way of pre-formatting these (rounding to a sensible number of decimal places, and adding a `%` sign if `data_type == percent`?

I considered doing the formatting inside the vue.js code, but then realised that generating a human-readable miniumum and maximum for a dataset feels like something the model should handle itself?

If `properties.minimum` and `properties.maximum` need to be maintained as they are, then maybe an _additional_ `properties.minimum_formatted` and `properties.maximum_formatted` (or similar) could be added instead?

Reply:
why, yes, yes there is.